### PR TITLE
Removing the injected repositories during the cloud-init phase once we finish that phase

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Card: https://github.com/SUSE/spacewalk/issues/19697

This PR clean-up all the injected repositories once the cloud-init phase finish.

It depends on: https://github.com/uyuni-project/sumaform/pull/1219

### CODE REVIEW

**Review by Bischoff — APPROVED** [review]
nice decoupling of responsibilities, thank you

maybe we also need something similar at the end of sumaform, not sure

**Review by rjmateus — COMMENTED** [review]
Just two comments

**Review by avshiliaev — APPROVED** [review]

File: backend_modules/libvirt/host/user_data.yaml [hunk 1]
```diff
@@ -183,9 +200,6 @@ packages: ["venv-salt-minion", "qemu-guest-agent"]
 %{ else }
 packages: ["qemu-guest-agent"]
 %{ endif }
-
-runcmd:
-  - zypper removerepo --all
```
[the remark this answers is not in the document]
Comment (srbarrios) [reply backend_modules/libvirt/host/user_data.yaml:188]: Not yet, but I will do.
On Monday I will continue my debugging related to Avahi packages.
When I deployed my SLES15SP5 env, using this branch, it failed in salt phase preparing avahi dependencies, apparently it was relying on the repos injected in cloud-init phase and I need to fix it. Once this is done, I will move to further investigate the reason why we decided to remove all kind of repositories from these images at cloud-init phase.

**Comment by srbarrios** [comment]
I'm not sure why these was closed by @maximenoel8.
I'm gonna open the PR again, because we might still want to integrate these code changes, but I will not be working on this now.

**Pull request closed without merging**